nifi-nar-bundles/nifi-extension-utils/nifi-dbcp-base/src/main/java/org/apache/nifi/dbcp/AbstractDBCPConnectionPool.java

```diff
@@ -157,7 +157,7 @@ private void loginKerberos(KerberosUser kerberosUser) throws InitializationExcep
 
     protected abstract DataSourceConfiguration getDataSourceConfiguration(final ConfigurationContext context);
 
-    protected void configureDataSource(final ConfigurationContext context, final DataSourceConfiguration configuration) {
+    protected void configureDataSource(final ConfigurationContext context, final BasicDataSource dataSource, final DataSourceConfiguration configuration) {
```

Comment:
Naming this variable `dataSource` shadows the protected member variable, so the property name should be changed for clarity.